### PR TITLE
Update android build to work with latest Plugins commit

### LINKED ISSUE #346: Updated CMake requirement may be problematic on Android
With the merge of the plugins branch, the CMake requirement was raised from CMake 3.10 to 3.12.

Android Studio only offers CMake 3.6, 3.10 and 3.18. Sadly the CMake version 3.18 offered by google is broken, so right now there is no way to compile the samples with an up-to-date Android studio.

I'm not sure how we should handle this. The easiest way would be dropping the CMake requirement back to 3.10, though I'm not sure if that's possible.

### PR DESCRIPTION
**Pull request by gpx1000**
## Description

Fix up the android project to allow it to work with latest.

Fixes #346

## General Checklist:

Please ensure the following points are checked:

- [ ] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [ ] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [ ] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
- [ ] I have commented any added functions (in line with Doxygen)
- [ ] I have commented any code that could be hard to understand
- [ ] My changes do not add any new compiler warnings
- [ ] My changes do not add any new validation layer errors or warnings
- [ ] I have used existing framework/helper functions where possible
- [ ] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [ ] My changes do not add any regressions
- [ ] I have tested every sample to ensure everything runs correctly
- [ ] This PR describes the scope and expected impact of the changes I am making

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [ ] I have tested the sample on at least one compliant Vulkan implementation
- [ ] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [ ] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [ ] Any dependent assets have been merged and published in downstream modules
- [ ] For new samples, I have added a paragraph with a summary to the appropriate chapter in the [samples readme](./../samples/README.md)

### CODE REVIEW

File: bldsys/cmake/template/gradle/build.gradle.in [hunk 1]
```diff
@@ -1,34 +1,31 @@
 buildscript {
     repositories {
-        jcenter()
+        mavenCentral()
```
Comment (TomAtkinsonArm) [inline bldsys/cmake/template/gradle/build.gradle.in:3]: Nice, I ran into this with our internal CI as well :+1:

**Comment by TomAtkinsonArm** [comment]
@gpx1000 was the android build broken? Or does this just clean up some of the android code in general?

Also does the gradle bump here need to be noted in any of the build documentation or the known issues section of the wiki?

**Comment by gpx1000** [comment]
Updating the gradle allows me to build with the cmake at version 3.12+ instead of having to roll back.  It should mean that the pr #347 is no longer necessary.

**Comment by TomAtkinsonArm** [comment]
> was the android build broken?

Sorry, didn't see the build failure in master before writing this comment

**Comment by gpx1000** [comment]
The only problem is getting the CI to play nicely with gradle.wrapper instead of insisting on using the built in gradle.  Working on figuring out a solution.

**Comment by SaschaWillems** [comment]
As usual, I'm having a hard time getting anything to build with all the broken Android native build tools. Can you post the versions you've been using?

I'm currently getting the following error when trying to build with gradle 7.x and NDK 23.x:

```
> Task :configureCMakeDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':configureCMakeDebug'.
> Cannot invoke "String.equals(Object)" because "line" is null
```

I don't have  a clue what's causing this, and haven't found a solution for the error message.

**Comment by gpx1000** [comment]
I'm going to go for a guess here: Your version of Java isn't happy.  That's definitely a Java error.  Admittedly a surprising one.

This is tested with openjdk11-jdk.  If you're on a more recent version of Java, lemme know.  Until 17 was released this month, 11 was the last LTS version; so it's reasonable to expect most wide support and stability for that version.

NDK r23
Android Studio: 2020.3.1 Patch 2

The rest gradle *should* figure out on its' own; but you're welcome to review the https://github.com/KhronosGroup/Vulkan-Samples/blob/master/.github/docker/Dockerfile for what versions are used in individual packages.  There's also https://github.com/KhronosGroup/Vulkan-Samples/blob/master/.github/docker/files/package-list.txt which is used to ensure the dependencies are downloaded.

**Comment by SaschaWillems** [comment]
Thanks for the hint with JDK11. It also seems my locally installed CMake version was interfering, and causing all the build troubles. After removing it from path, the build is now working.

**Comment by gpx1000** [comment]
Glad to hear it Sascha!  I was getting worried I might have broken something important :)

Next challenge will be to document the right versions and what to install etc.  I figured we can get this merged first so the CI is happy for all of the other PRs then come back to update documentation.

**Comment by SaschaWillems** [comment]
The project now builds fine for me. Both from the command line via gradle and from Android Studio with all the latest components. I did a clean install to get a default Android Studio setup (including CMake 3.18) and with this PR I could again compile the project out of the box. Great job :)

As for the documentation, we have this [wiki page](https://github.com/KhronosGroup/Vulkan-Samples/wiki/Known-Issues) we could update, and then just refer to it from the build readme for Android?

**Review by SaschaWillems — APPROVED** [review]

**Comment by TomAtkinsonArm** [comment]
Looking good, on my to-do to test locally today and merge ASAP

**Comment by TomAtkinsonArm** [comment]
I'm having a few issues when building locally. Does this change require Android Studio 2020.3.1 Patch 2 and above?

**Comment by gpx1000** [comment]
It shouldn't require a specific version of Android studio.  Gradlew is designed to go get the version of gradle and dependencies if they don't exist.  All that it could require is the current version of the ndk.  Which that's probably why Sascha had problems with his cmake.
What problems are you getting?

**Comment by TomAtkinsonArm** [comment]
Currently NDK not found, I have version 23 installed through SDK Tools

**Comment by gpx1000** [comment]
If you only have the SDK tools installed; then that's likely the issue.  It's kinda a hack these days to have systems that don't have the full Android SDK available.  Thus the normal way that things work:
the SDK will install the NDK and install it under the SDK folder.  gradle uses this knowledge to discover where the NDK exists.  The SDK location is determined in local.properties at your root level like this:
sdk.dir = 

If the NDK exists somewhere other than under the SDK, then it needs to be listed in the local.properties file:
ndk.dir =

NB: setting the NDK path like this is considered deprecated and you will get warnings about that.  The assumption is transitioning to the ndk will be found in Sdk/ndk-bundle or Sdk/ndk.
also note that this is the way that we got the CI system to work so you can review the method of setting the local.properties there.

**Comment by TomAtkinsonArm** [comment]
Looks like I need to explicitly set `ndk.dir` to get this to build

The next failure is:

```
> Task :configureCMakeRelWithDebInfo FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':configureCMakeRelWithDebInfo'.
> Cannot invoke "String.equals(Object)" because "line" is null

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug                                                                                                                      option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 13s
11 actionable tasks: 11 executed

$ cmake --version
cmake version 3.21.2

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

I may need to attempt a complete reinstall similar to Sascha 👀

Although it seems broken if the build system can not work with a system install of CMake??

**Comment by gpx1000** [comment]
I'm not entirely sure why that happens to be honest.  You can use the system cmake with gradle builds as that feature was introduced in Oreo timeframe if memory serves.
However the build.gradle must specify using the system cmake install instead of the one found in the ndk via the normal Android studio install process.  My best guess is there's some kind of conflict happening where the cmake that the gradle is using in the ndk is selecting the tooling in the system path.  It might be enough to refresh the system cmake.  However this is a deep bug somewhere that I'm unfamiliar with so it's just a guess.

**Comment by SaschaWillems** [comment]
@TomAtkinsonArm I had the same error. For me, it was caused by having a very recent CMake version in my path (which seems to be the case for you too, judging by the error message). Gradle would then pick that up and fail with that exact same message. After removing that CMake from my path, it used the one installed with the SDK/NDK and worked as expected.

**Edit:** One problem with removing CMake from your path: The .bat to generate the android build files doesn't work anymore as it requires CMake. Not sure how we can handle it so that generating the scripts and building properly works :/

**Comment by gpx1000** [comment]
If you have a system cmake install in your path happen after the android studio install do you still have the conflict problem @SaschaWillems ?

**Comment by SaschaWillems** [comment]
Yes, I reinstalled Android Studio multiple times, and It only worked after removing CMake from my path. There is a note in google's pages that this might work with CMake 3.21+ (I'm still on 3.20), so it might work with a more recent version.

**Pull request merged**